visual-dotnet/SauceLabs.Visual/VisualClient.cs

```diff
@@ -79,10 +81,16 @@ public static async Task<VisualClient> Create(WebDriver wd, Region region, strin
 
         private async Task SetupBuild(CreateBuildOptions buildOptions)
         {
-            var response = await this._api.WebDriverSessionInfo(_jobId, _sessionId);
+            var response = await _api.WebDriverSessionInfo(_jobId, _sessionId);
             var metadata = response.EnsureValidResponse();
             _sessionMetadataBlob = metadata.Result.Blob;
 
+            if (_sharedBuild != null)
```

Comment:
I think we should hide what exactly is under the shared build by using some kind of abstraction - why? I think we should keep the client code as clean as possible.


How about moving this code to BuildFactory responsible for generating/managing build?

Reply:
You're right. This will also allow to have a build that is shared but that could still be isolated by region (which is not the case yet).

In a typical usecase that should not happen, but that would be a flaw.